### PR TITLE
Add functionality for updating `.yarnrc.yml` file

### PR DESCRIPTION
**Pull request by Mrtenz**
## Description

This pull request adds new functionality to update the local `.yarnrc.yml` file to match the module template. It enables
the updating from Yarn 1 to Yarn 3 and to a more recent version of Yarn 3. It also includes the installation of any plugins
specified in the module template.

This change also allows for updates to Yarn to be applied and new plugins to be installed simply by updating the
module template. This helps us keep up-to-date with the latest Yarn releases and improve project development
efficiency.

## Changes

1. Add functionality for updating the `.yarnrc.yml` file to match the module template.
2. Support updating from Yarn 1 to Yarn 3 and updating to a more recent version of Yarn 3.
3. Install any plugins specified in the module template.

### CODE REVIEW

**Comment by Mrtenz** [comment]
Blocked by #3.

File: src/processing/yarnrc.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  const { plugins } = parsedTemplateYarnRc;
+  assert(Array.isArray(plugins));
+
+  // To avoid errors with the Yarn path not existing, we need to temporarily
+  // delete it from the configuration. The plugins may not be available
+  // either, so we need to delete them as well, and install them later.
+  delete parsedTemplateYarnRc.yarnPath;
+  delete parsedTemplateYarnRc.plugins;
+
+  await writeFile(CURRENT_YARN_PATH, dump(parsedTemplateYarnRc));
```
Comment (FrederikBolding) [inline src/processing/yarnrc.ts:68]: Does this correctly overwrite an existing file? I thought you needed to specify a flag for that

File: src/processing/yarnrc.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+  const newVersion = semver.lt(currentYarnVersion, templateYarnVersion)
```
Comment (FrederikBolding) [inline src/processing/yarnrc.ts:70]: This doesn't match the comment at the top of the function does it? In this case we would not downgrade if our current version is higher?

File: src/processing/yarnrc.ts (hunk 1 above)
Comment (Mrtenz) [reply src/processing/yarnrc.ts:68]: It does.

File: src/processing/yarnrc.ts (hunk 2 above)
Comment (Mrtenz) [reply src/processing/yarnrc.ts:70]: > - If the Yarn 3 version is newer than the template version, a warning will be
displayed. The `.yarnrc.yml` is updated to use the current version.

By that I mean that the `.yarnrc.yml` will use the format of `metamask-module-template`, but use the most recent Yarn version.

Comment (FrederikBolding) [reply src/processing/yarnrc.ts:70]: Can we make that clearer?

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**